packages/dev/core/src/Maths/math.path.ts

```diff
@@ -96,17 +96,18 @@ export class Angle {
 
     /**
      * Gets the angle between the two vectors
-     * @param a defines first point as the origin
-     * @param b defines point
+     * @param a defines first vector
+     * @param b defines vector
      * @returns Returns an new Angle between 0 and PI
      */
     public static BetweenTwoVectors<Vec extends Vector2 | Vector3 | Vector4>(a: DeepImmutable<Vec>, b: DeepImmutable<Vec>): Angle {
-        const na = a.normalizeToNew() as Vec;
-        const nb = b.normalizeToNew() as Vec;
-        let cosv = na.dot(nb as any);
-        cosv = Scalar.Clamp(cosv, -1, 1);
-        const theta = Math.acos(cosv);
-        return new Angle(theta);
+        let product = a.lengthSquared() * b.lengthSquared();
+        if (product === 0) return new Angle(Math.PI / 2);
+        product = Math.sqrt(product);
+        let cosVal = (a.constructor as any).Dot(b) / product;

```

Comment:
is `any` needed here? dont all Vec types have this method defined?

Reply:
This was written before my pr passed, and not all vectors had dot methods at the time.

Now I've changed it around

Reply:
But still the any is left. That's what I'm trying to understand - why is it needed